### ISSUE TITLE
scraping with python, node.js or go failed because miss $ in --data-binary

### ISSUE THREAD

**Issue by goldalworming** [NickCarneiro/curlconverter#185]
please focus on data-binary $'[{"query":"query getD..., in normal request there isn't $ before ' in this curl 

	curl 'https://www.sorabel.com/graphql/v2/?accessToken=SS-JWS-1%20eyJhbGciOiJIUzI1NiJ9.eyJzTEkiOnRydWUsInNNRUkiOiJlYmJiNjM1Mi03YTllLTQxNTItODg3YS1hY2Y2MGQ0Y2FmNTkiLCJleHAiOjE1NzY2MDM5MzEsImlhdCI6MTU3NTM5NDMzMSwic0kiOiJhOWZlMGE0MS05NGMxLTRiZmYtOTNkYy00MTg1ZjRjOGQ5NGUifQ.XCYoJ70d1hQW2axGDVn__ODyQAK6gEwQ_-ipppcVgjI&bundleVersion=292cc6e&freeDeliveryBucket' -H 'authority: www.sorabel.com' -H 'accept: */*' -H 'origin: https://www.sorabel.com' -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36' -H 'x-request-id: a9874c49-5200-4919-974d-30133f594af6' -H 'content-type: text/plain' -H 'sec-fetch-site: same-origin' -H 'sec-fetch-mode: cors' -H 'referer: https://www.sorabel.com/products/category/trend-red-outfit' -H 'accept-encoding: gzip, deflate, br' -H 'accept-language: en-US,en;q=0.9,id;q=0.8,ms;q=0.7' -H 'cookie: visid_incap_1886739=gD/fYB5yQ8KxdvO4KTmTYdaO9FwAAAAAQUIPAAAAAABpxlYSWF7yAguuPnL1rY+f; 4261f087-09bf-405a-8922-910596c5cdb7-latestReadOffset=1567495152746; _gcl_au=1.1.2056193586.1571034001; _ga=GA1.2.1220360820.1571034001; _fbp=fb.1.1571034001401.397232614; cto_lwid=de457af4-3a9c-4714-85cc-2e3da7aa7130; orderCount=1; ebbb6352-7a9e-4152-887a-acf60d4caf59-latestReadOffset=1574479107475; __ss_tci_jws=eyJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2MDY5NTEyNTcsImlhdCI6MTU3NTM5NDMzMSwiY29va2llSWQiOiJjMWM1NzU3MS05ZDEyLTQ1ZWItOGYxYS0wMDdmNTkyYjVlMzIifQ.G5Oe_XeVGYwv4whwYBCGiII3uh5_x3Cplcq9yXMYxCo; __ss_tsi=fdfe5b50-f6a9-4334-9b6c-1c2372ad589e; token=SS-JWS-1%20eyJhbGciOiJIUzI1NiJ9.eyJzTEkiOnRydWUsInNNRUkiOiJlYmJiNjM1Mi03YTllLTQxNTItODg3YS1hY2Y2MGQ0Y2FmNTkiLCJleHAiOjE1NzY2MDM5MzEsImlhdCI6MTU3NTM5NDMzMSwic0kiOiJhOWZlMGE0MS05NGMxLTRiZmYtOTNkYy00MTg1ZjRjOGQ5NGUifQ.XCYoJ70d1hQW2axGDVn__ODyQAK6gEwQ_-ipppcVgjI; incap_ses_1112_1886739=uTbWRVpDoRaA41J6oJ5uDxuc5l0AAAAAg5k2uHaExG1VIpBnC51wWA==; __ss_tsi_n=835d40fd-e382-4356-96b1-e990e2028d4c; lastActiveTime=1575394387.239' --data-binary $'[{"query":"query getDynamicContentPage($app: Boolean, $preview: Boolean, $previewSource: String, $slug: String\u0021, $algo: String, $widgetOffset: Int, $contentOffset: Int, $deviceType: String\u0021, $cookieId: String\u0021, $title: String, $categories: String, $category_ids: String, $motives: String, $materials: String, $models: String, $brands: String, $colors: String, $std_colors: String, $sizes: String, $tags: String, $genders: String, $ages: String, $priceFrom: String, $priceTo: String, $order: String, $sort: String) {\\n  dynamic(app: $app, preview: $preview, previewSource: $previewSource, slug: $slug, algo: $algo, widgetOffset: $widgetOffset, contentOffset: $contentOffset, deviceType: $deviceType, cookieId: $cookieId, title: $title, categories: $categories, category_ids: $category_ids, motives: $motives, materials: $materials, models: $models, brands: $brands, colors: $colors, std_colors: $std_colors, sizes: $sizes, tags: $tags, genders: $genders, ages: $ages, priceFrom: $priceFrom, priceTo: $priceTo, order: $order, sort: $sort) {\\n    slug\\n    params\\n    widgetOffset\\n    contentOffset\\n    nextWidgetOffset\\n    nextContentOffset\\n    metadata {\\n      title\\n      description\\n      seoTitle\\n      seoDescription\\n      image\\n      isShareable\\n      noIndex\\n      commentIndex\\n      header\\n      __typename\\n    }\\n    content {\\n      id\\n      type\\n      offset\\n      nextContentOffset\\n      data {\\n        appearance\\n        actionLink\\n        actionTitle\\n        title\\n        context {\\n          widgetType\\n          type\\n          name\\n          widgetPosition\\n          widgetId\\n          contextDataPath\\n          additionalData\\n          __typename\\n        }\\n        layout {\\n          ...LayoutFragment\\n          __typename\\n        }\\n        __typename\\n      }\\n      content {\\n        ...SupportedWidgetFragment\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n\\nfragment SupportedWidgetFragment on BaseWidget {\\n  id\\n  offset\\n  ... on ActionTextWidget {\\n    ...ActionTextWidgetFragment\\n    __typename\\n  }\\n  ... on BannerWidget {\\n    ...BannerWidgetFragment\\n    __typename\\n  }\\n  ... on DownloadAppWidget {\\n    ...DownloadAppWidgetFragment\\n    __typename\\n  }\\n  ... on EditorialWidget {\\n    ...EditorialWidgetFragment\\n    __typename\\n  }\\n  ... on ImageTextWidget {\\n    ...ImageTextWidgetFragment\\n    __typename\\n  }\\n  ... on ImageWidget {\\n    ...ImageWidgetFragment\\n    __typename\\n  }\\n  ... on ListHeaderWidget {\\n    ...ListHeaderWidgetFragment\\n    __typename\\n  }\\n  ... on ListItemWidget {\\n    ...ListItemWidgetFragment\\n    __typename\\n  }\\n  ... on NotificationWidget {\\n    ...NotificationWidgetFragment\\n    __typename\\n  }\\n  ... on PageShareWidget {\\n    ...PageShareWidgetFragment\\n    __typename\\n  }\\n  ... on ProductWidget {\\n    ...ProductWidgetFragment\\n    __typename\\n  }\\n  ... on PagePostWidget {\\n    ...PagePostWidgetFragment\\n    __typename\\n  }\\n  ... on TabWidget {\\n    ...TabWidgetFragment\\n    __typename\\n  }\\n  ... on TextWidget {\\n    ...TextWidgetFragment\\n    __typename\\n  }\\n  ... on USPWidget {\\n    ...USPWidgetFragment\\n    __typename\\n  }\\n  ... on VideoWidget {\\n    ...VideoWidgetFragment\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment ActionTextWidgetFragment on ActionTextWidget {\\n  type\\n  data {\\n    appearance\\n    actionTitle\\n    actionLink\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment LayoutFragment on WidgetLayout {\\n  columns\\n  rows\\n  columnSize\\n  rowSize\\n  marginTop\\n  marginRight\\n  marginBottom\\n  marginLeft\\n  paddingTop\\n  paddingRight\\n  paddingBottom\\n  paddingLeft\\n  __typename\\n}\\n\\nfragment BannerWidgetFragment on BannerWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment DownloadAppWidgetFragment on DownloadAppWidget {\\n  type\\n  data {\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment EditorialWidgetFragment on EditorialWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      actionTitle\\n      title\\n      slug\\n      image {\\n        fullUrl\\n        width\\n        height\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment ImageTextWidgetFragment on ImageTextWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      title\\n      link\\n      image {\\n        fullUrl(maxWidth: 480)\\n        width\\n        height\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment ImageWidgetFragment on ImageWidget {\\n  type\\n  data {\\n    appearance\\n    actionLink\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      fullUrl(maxWidth: 480)\\n      width\\n      height\\n      title\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment ListHeaderWidgetFragment on ListHeaderWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      title\\n      image {\\n        fullUrl(width: 330)\\n        width\\n        height\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment ListItemWidgetFragment on ListItemWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      title\\n      content\\n      link\\n      image {\\n        fullUrl\\n        width\\n        height\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment NotificationWidgetFragment on NotificationWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment PageShareWidgetFragment on PageShareWidget {\\n  type\\n  data {\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment ProductWidgetFragment on ProductWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      id\\n      slug\\n      title\\n      quantity\\n      tryOutfit\\n      tags\\n      code\\n      brand\\n      category\\n      material\\n      materialDescription\\n      favouriteCount\\n      favourite\\n      isPublished\\n      discountPercentage\\n      normalPrice {\\n        amount\\n        currency\\n        __typename\\n      }\\n      discountPrice {\\n        amount\\n        currency\\n        __typename\\n      }\\n      price {\\n        amount\\n        currency\\n        __typename\\n      }\\n      salePrice {\\n        amount\\n        currency\\n        __typename\\n      }\\n      variants {\\n        color\\n        id\\n        quantity\\n        size\\n        sku\\n        title\\n        __typename\\n      }\\n      images {\\n        fullUrl(width: 480)\\n        height\\n        width\\n        filePath\\n        tags\\n        __typename\\n      }\\n      detailMarkdown\\n      descriptionMarkdown\\n      modelPhotoColor\\n      modelPhotoSize\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment PagePostWidgetFragment on PagePostWidget {\\n  type\\n  data {\\n    appearance\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      title\\n      pageSlug\\n      description\\n      publishedAt\\n      image {\\n        fullUrl(maxWidth: 480)\\n        width\\n        height\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment TabWidgetFragment on TabWidget {\\n  type\\n  data {\\n    appearance\\n    actionTitle\\n    actionLink\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment TextWidgetFragment on TextWidget {\\n  type\\n  data {\\n    appearance\\n    actionTitle\\n    actionLink\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      title\\n      content\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment USPWidgetFragment on USPWidget {\\n  type\\n  data {\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n\\nfragment VideoWidgetFragment on VideoWidget {\\n  type\\n  data {\\n    context {\\n      type\\n      widgetType\\n      name\\n      widgetPosition\\n      widgetId\\n      contextDataPath\\n      additionalData\\n      __typename\\n    }\\n    layout {\\n      ...LayoutFragment\\n      __typename\\n    }\\n    entity {\\n      content\\n      __typename\\n    }\\n    __typename\\n  }\\n  __typename\\n}\\n","variables":{"slug":"products/category/trend-red-outfit","widgetOffset":1,"contentOffset":27,"deviceType":"desktop","cookieId":"c1c57571-9d12-45eb-8f1a-007f592b5e32","app":false},"operationName":"getDynamicContentPage"}]' --compressed

how do I translate $'... in --data-binary??

**Comment by goldalworming** [comment]
another example of $ in data binary

    curl 'https://gql.tokopedia.com/' --data-binary '[{"operationName":"PDPShopInfoQuery","variables":{"domain":"umproduktion","fields":["allow_manage","assets","core","favorite","last_active","other-shiploc","other-goldos","status","closed_info"]},"query":"query PDPShopInfoQuery($fields: [String\u0021]\u0021, $domain: String) {\\n  shopInfo: shopInfoByID(input: {shopIDs: [0], fields: $fields, domain: $domain}) {\\n    result {\\n      favoriteData {\\n        alreadyFavorited\\n        __typename\\n      }\\n      goldOS {\\n        isGold\\n        isGoldBadge\\n        isOfficial\\n        badgeSVG\\n        title\\n        __typename\\n      }\\n      isAllowManage\\n      shopLastActive\\n      location\\n      shippingLoc {\\n        districtName\\n        cityName\\n        __typename\\n      }\\n      shopAssets {\\n        avatar\\n        __typename\\n      }\\n      shopCore {\\n        description\\n        domain\\n        shopID\\n        name\\n        url\\n        __typename\\n      }\\n      statusInfo {\\n        shopStatus\\n        statusMessage\\n        statusTitle\\n        __typename\\n      }\\n      closedInfo {\\n        closedNote\\n        until\\n        reason\\n        __typename\\n      }\\n      __typename\\n    }\\n    error {\\n      message\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n"},{"operationName":"ProductVariantQuery","variables":{"productID":"423367103","includeCampaign":true},"query":"query ProductVariantQuery($productID: String\u0021, $includeCampaign: Boolean\u0021) {\\n  getProductVariant(productID: $productID, option: {userID: \\"0\\", includeCampaign: $includeCampaign}) {\\n    parentID\\n    defaultChild\\n    variant {\\n      productVariantID\\n      variantID\\n      variantUnitID\\n      name\\n      identifier\\n      unitName\\n      position\\n      option {\\n        productVariantOptionID\\n        variantUnitValueID\\n        value\\n        hex\\n        picture {\\n          urlOriginal: url\\n          urlThumbnail: url200\\n          __typename\\n        }\\n        __typename\\n      }\\n      __typename\\n    }\\n    children {\\n      productID\\n      price\\n      priceFmt\\n      sku\\n      optionID\\n      productName\\n      productURL\\n      picture {\\n        urlOriginal: url\\n        urlThumbnail: url200\\n        __typename\\n      }\\n      stock {\\n        stock\\n        isBuyable\\n        alwaysAvailable\\n        isLimitedStock\\n        stockWording\\n        stockWordingHTML\\n        otherVariantStock\\n        minimumOrder\\n        maximumOrder\\n        __typename\\n      }\\n      isCOD\\n      isWishlist\\n      campaignInfo {\\n        stock\\n        originalStock\\n        endDateUnix\\n        isActive\\n        appLinks\\n        startDate\\n        campaignID\\n        isAppsOnly\\n        campaignType\\n        originalPrice\\n        discountPrice\\n        originalPriceFmt\\n        discountPriceFmt\\n        campaignTypeName\\n        discountPercentage\\n        hideGimmick\\n        __typename\\n      }\\n      __typename\\n    }\\n    sizeChart\\n    enabled\\n    alwaysAvailable\\n    stock\\n    __typename\\n  }\\n}\\n"},{"operationName":"PDPInfoQuery","variables":{"shopDomain":"umproduktion","productKey":"sepatu-kickers-casual-saka-pria-terlaris-promo-sepatu-terbaru-kekinian-hitam-39"},"query":"query PDPInfoQuery($shopDomain: String, $productKey: String) {\\n  getPDPInfo(productID: 0, shopDomain: $shopDomain, productKey: $productKey) {\\n    stats {\\n      countView\\n      countReview\\n      countTalk\\n      rating\\n      __typename\\n    }\\n    txStats {\\n      txSuccess\\n      txReject\\n      itemSold\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n"},{"operationName":"ShopCommitmentQuery","variables":{"shopID":"5558124","price":156000},"query":"query ShopCommitmentQuery($shopID: String\u0021, $price: Int) {\\n  shopCommitment(shopID: $shopID, itemPrice: $price) {\\n    result {\\n      iconURL\\n      isNowActive\\n      staticMessages {\\n        pdpMessage\\n        __typename\\n      }\\n      __typename\\n    }\\n    error {\\n      message\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n"}]' --compressed

**Comment by rajun274** [comment]
Hello!

I'm having the same problem related to dollar sign $ prefixed to the data-binary value.

I had used "Copy as cURL" from Google Dev Tools. The data-binary portion of the cURL command has the following:

--data-binary $'[{"keyPart":"valuePart",<rest_is_omitted>

The Python code created by your tool doesn't create the proper POST command, and so I get an error from the server I'm hitting. I fixed my code in the following way:

import json
data = [{'keyPart': 'valuePart'}]
requests.post(url, data=json.dumps(data)) # omitting the "json.dumps" breaks the HTTP request

Thanks!
